conda/core/path_actions.py

```diff
@@ -490,6 +530,79 @@ def reverse(self):
+            py_files = tuple((axn.target_short_path for axn in file_link_actions
+                              if noarch_py_file_re.match(axn.source_short_path)))
+            pyc_files = tuple((pyc_path(pf, py_ver) for pf in py_files))
+            return (cls(transaction_context, package_info, target_prefix, py_files, pyc_files), )
+        else:
+            return ()
+
+    def __init__(self, transaction_context, package_info, target_prefix,
+                 source_short_paths, target_short_paths):
+        self.transaction_context = transaction_context
+        self.package_info = package_info
+        self.target_prefix = target_prefix
+        self.source_short_paths = source_short_paths
+        self.target_short_paths = target_short_paths
+        self.prefix_path_data = None
+        self.prefix_paths_data = [
+            PathDataV1(_path=p, path_type=PathType.pyc_file,) for p in self.target_short_paths]
+        self._execute_successful = False
+
+    @property
+    def target_full_paths(self):
+        def join_or_none(prefix, short_path):
+            if prefix is None or short_path is None:
+                return None
+            else:
+                return join(prefix, win_path_ok(short_path))
+        return (join_or_none(self.target_prefix, p) for p in self.target_short_paths)
+
+    @property
+    def source_full_paths(self):
+        def join_or_none(prefix, short_path):
+            if prefix is None or short_path is None:
+                return None
+            else:
+                return join(prefix, win_path_ok(short_path))
+        return (join_or_none(self.target_prefix, p) for p in self.source_short_paths)
+
+    def verify(self):
+        self._verified = True
+
+    def cleanup(self):
+        # create actions typically won't need cleanup
+        pass
+
+    def execute(self):
+        # compile_pyc is sometimes expected to fail, for example a python 3.6 file
+        #   installed into a python 2 environment, but no code paths actually importing it
+        # technically then, this file should be removed from the manifest in conda-meta, but
+        #   at the time of this writing that's not currently happening
+        log.trace("compiling %s", ' '.join(self.target_full_paths))
+        target_python_version = self.transaction_context['target_python_version']
+        python_short_path = get_python_short_path(target_python_version)
+        python_full_path = join(self.target_prefix, win_path_ok(python_short_path))
+        compile_multiple_pyc(python_full_path, self.source_full_paths, self.target_full_paths)
+        self._execute_successful = True
+
+    def reverse(self):
+        # this removes all pyc files even if they were not created
+        if self._execute_successful:
+            log.trace("reversing pyc creation %s", self.target_full_path)
+            for target_full_path in self.target_full_paths:
+                rm_rf(target_full_path)
```

Comment:
calling rm_rf one by one is probably slow here, but I'm not sure there's a good way around that right now.  If it is isolated as a bottleneck down the line, it may be better to collect up filenames and batch calls to the native shell rm utility.

Reply:
Single file rm_rf was used it the original implementation and for non-noarch packages.  I do not expect this loop to be much of a bottleneck.